x-pack/plugin/ent-search/src/main/java/org/elasticsearch/xpack/application/connector/ConnectorIngestPipeline.java

```diff
@@ -0,0 +1,142 @@
+    }
+
+    public ConnectorIngestPipeline(StreamInput in) throws IOException {
+        this.extractBinaryContent = in.readBoolean();
+        this.name = in.readString();
+        this.reduceWhitespace = in.readBoolean();
+        this.runMlInference = in.readBoolean();
+    }
+
+    public static final ParseField EXTRACT_BINARY_CONTENT_FIELD = new ParseField("extract_binary_content");
+    public static final ParseField NAME_FIELD = new ParseField("name");
+    public static final ParseField REDUCE_WHITESPACE_FIELD = new ParseField("reduce_whitespace");
+    public static final ParseField RUN_ML_INFERENCE_FIELD = new ParseField("run_ml_inference");
+
+    private static final ConstructingObjectParser<ConnectorIngestPipeline, Void> PARSER = new ConstructingObjectParser<>(
+        "connector_ingest_pipeline",
+        true,
+        args -> new Builder().setExtractBinaryContent((Boolean) args[0])
+            .setName((String) args[1])
+            .setReduceWhitespace((Boolean) args[2])
+            .setRunMlInference((Boolean) args[3])
+            .createConnectorIngestPipeline()
+    );
+
+    static {
+        PARSER.declareBoolean(constructorArg(), EXTRACT_BINARY_CONTENT_FIELD);
+        PARSER.declareString(constructorArg(), NAME_FIELD);
+        PARSER.declareBoolean(constructorArg(), REDUCE_WHITESPACE_FIELD);
+        PARSER.declareBoolean(constructorArg(), RUN_ML_INFERENCE_FIELD);
+    }
+
+    public static ConnectorIngestPipeline fromXContentBytes(BytesReference source, XContentType xContentType) {
+        try (XContentParser parser = XContentHelper.createParser(XContentParserConfiguration.EMPTY, source, xContentType)) {
+            return ConnectorIngestPipeline.fromXContent(parser);
+        } catch (IOException e) {
+            throw new ElasticsearchParseException("Failed to parse: " + source.utf8ToString(), e);
+        }
+    }
+
+    public static ConnectorIngestPipeline fromXContent(XContentParser parser) throws IOException {
+        return PARSER.parse(parser, null);
+    }
+
+    @Override
+    public XContentBuilder toXContent(XContentBuilder builder, Params params) throws IOException {
+        builder.startObject();
+        {
+            builder.startObject("properties");
+            {
+                builder.field(EXTRACT_BINARY_CONTENT_FIELD.getPreferredName(), extractBinaryContent);
+                if (name != null) {
+                    builder.field(NAME_FIELD.getPreferredName(), name);
+                }
+                builder.field(REDUCE_WHITESPACE_FIELD.getPreferredName(), reduceWhitespace);
+                builder.field(RUN_ML_INFERENCE_FIELD.getPreferredName(), runMlInference);
+            }
+            builder.endObject();
+        }
+        builder.endObject();
+        return builder;
+    }
+
+    @Override
+    public void writeTo(StreamOutput out) throws IOException {
+        out.writeBoolean(extractBinaryContent);
+        out.writeOptionalString(name);
```

Comment:
I think you want `writeString` here

Reply:
Good catch, all args are required here, confirmed it.